docs: update contributing docs (#13743)
open-telemetry/opentelemetry-java-instrumentation: CONTRIBUTING.md

Before submitting new features or changes to current functionality, it is recommended to first
[open an issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/new)
and [discuss](https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/new?category=q-a) your ideas or propose the changes you wish to make.

## Building

> This project requires Gradle 8.13 and Java 21 to build and run tests. Newer Gardle and JDK may work, but this version is used in CI.

Reviewer: ```suggestion
This project requires Gradle 8.13+ and Java 21 to build and run tests. Newer Gradle and JDK versions may work, but these are the versions used in CI.
```

Before submitting new features or changes to current functionality, it is recommended to first
[open an issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/new)

> and [discuss](https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/new?category=q-a) your ideas or propose the changes you wish to make.

Reviewer 2: To me the intent of `it is recommended to first open an issue and discuss your ideas` is that you open an issue and that issue is used for discussion. Linking the `discuss` to opening a discussion in that sentence does, in my opinion, not make sense as it leaves an impression that you should open both an issue and a discussion.
